### PR TITLE
samples/gocdk-secrets: add test

### PR DESCRIPTION
**Pull request by jba**

### CODE REVIEW

File: samples/gocdk-secrets/main.go [hunk 1]
```diff
@@ -31,8 +31,8 @@ import (
 	_ "gocloud.dev/secrets/awskms"
 	_ "gocloud.dev/secrets/azurekeyvault"
 	_ "gocloud.dev/secrets/gcpkms"
-	_ "gocloud.dev/secrets/hashivault"
 	_ "gocloud.dev/secrets/localsecrets"
+	_ "gocloud.dev/secrets/vault"
```
Comment (vangent) [inline samples/gocdk-secrets/main.go]: ? The package is named `hashivault`.

File: samples/go.mod [hunk 2]
```diff
@@ -20,6 +20,7 @@ require (
 	gocloud.dev/pubsub/rabbitpubsub v0.15.0
 	gocloud.dev/runtimevar/etcdvar v0.15.0
 	gocloud.dev/secrets/hashivault v0.15.0
+	gocloud.dev/secrets/vault v0.15.0
```
Comment (vangent) [inline samples/go.mod]: I don't think there should be any `go.mod` or `go.sum` changes in this PR, and I'm not sure where `gocloud.dev/secrets/vault` is coming from.

File: samples/gocdk-secrets/main.go (hunk 1 above)
Comment (jba) [reply samples/gocdk-secrets/main.go]: done

File: samples/go.mod (hunk 2 above)
Comment (jba) [reply samples/go.mod]: done

**Comment by jba** [comment]
PTAL

**Pull request merged**